### PR TITLE
Display Golem startup messages in status bar (Network tab)

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
**Running**
- build and replace Golem binaries from the following branch: golemfactory/golem#1227

**TODO**
- [x] Disable the Start / Stop Golem button until Golem is in "Ready" state
- [ ] Paraphrase status messages

### CODE REVIEW

File: src/reducers/realTime.js [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
     footerInfo: null
 }
 
 let badgeActive = false
 let badgeTemp = 0
 
+
+function nodesString(num) {
+    if (num < 1) return 'No Nodes Connected';
+    let postfix = num != 1 ? 's' : '';
```
Comment (maaktweluit) [inline src/reducers/realTime.js]: When possible i try to:
- use `const` over `let`
- use `!==` over `!=`

Both could apply here, but are very minor points.

File: src/reducers/realTime.js [hunk 2]
```diff
@@ -39,7 +51,20 @@ const realTime = (state = initialState, action) => {
 
     case SET_CONNECTED_PEERS:
         return Object.assign({}, state, {
-            connectedPeers: action.payload
+            connectedPeers: action.payload,
+            golemStatus: {
+                status: 'Ready',
```
Comment (maaktweluit) [inline src/reducers/realTime.js:56]: Could use a ENUM to not hard-code the status strings, not trivial, maybe follow up issue?

File: src/sagas/golem.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+function dig(src, ...rest) {
+    for ( let arg of rest )
+        src = src[arg];
+    return src;
+}
+
+
+function getGolemStatus(component, method, stage, data) {
+    let result = {};
```
Comment (maaktweluit) [inline src/sagas/golem.js]: since result is never re-assigned `const` is also possible here.

File: src/sagas/golem.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+        return () => _handleUNSUBPUB(on_status, session, config.GOLEM_STATUS_CH);
+    })
+}
+
+
+/**
+ * [*golemStatus generator]
+ * @param  {Object} session     [Websocket connection session]
+ * @yield   {Object}            [Action object]
```
Comment (maaktweluit) [inline src/sagas/golem.js]: space too much? before `{Object}`, it looks like the formatting shows all grey due to this.

File: src/sagas/handler.js [hunk 5]
```diff
@@ -84,7 +84,7 @@ export let config = Object.freeze({
     //Task Presets
     TASK_PRESETS_RPC: 'comp.tasks.preset.get',
     SAVE_TASK_PRESET_RPC: 'comp.tasks.preset.save',
-    DELETE_TASK_PRESET_RPC: 'comp.tasks.preset.delete'
+    DELETE_TASK_PRESET_RPC: 'comp.tasks.preset.delete',
```
Comment (maaktweluit) [inline src/sagas/handler.js]: this extra `,` makes it a non valid JSON, but Javascript will still eat it.

File: src/components/network/index.js [hunk 6]
```diff
@@ -208,10 +214,10 @@ export class MainFragment extends React.Component {
             {managePresetModal && <ManagePresetModal closeModal={::this._closeModal}/>}
             <div className="section__actions">
                 <div className="section__actions-status">
-                    <span className={`icon-status-dot ${!connectionProblem ? 'icon-status-dot--active' : 'icon-status-dot--warning'}`}/>
-                    <span>{!connectionProblem ? `${connectedPeers} ${connectedPeers > 1 ? 'Nodes' : 'Node'}` : 'No Connection' }</span>
+                    <span className={`icon-status-dot ${!connectionProblem && isGolemReady(golemStatus.status) ? 'icon-status-dot--active' : 'icon-status-dot--warning'}`}/>
```
Comment (maaktweluit) [inline src/components/network/index.js:217]: Could repeat a bit less:
```
`icon-status-dot icon-status-dot--${!connectionProblem && isGolemReady(golemStatus.status) ? 'active' : 'warning'}`
```
Or even:
```
`icon-status-dot icon-status-dot--${_getIconModifierByStatus( connectionProblem, golemStatus)}`
```

**Comment by maaktweluit** [comment]
Hey @mfranciszkiewicz,

Since there was a review requested I had a look. I'm new to the project so not aware of what standards and practices you follow. Just commented all i could think of, all where non breaking comments but maybe they help.

Have a good one!

Maaktweluit

**Review by maaktweluit — APPROVED** [review]

**Review by banasiakadam60 — APPROVED** [review]

**Pull request merged**